Xamarin.Forms.Platform.Android/AppCompat/Platform.cs

```diff
@@ -294,6 +294,9 @@ void SetPageInternal(Page newRoot)
 				foreach (var root in _navModel.Roots)
 					renderersToDispose.Add(Android.Platform.GetRenderer(root));
 
+				if (Android.Platform.GetRenderer(Page) != null)
+					Page.ClearValue(Android.Platform.RendererProperty);
```

Comment:
and you might want to dispose the renderer too

Reply:
The renderer is disposed by the Cleanup method called later in this method.